x-pack/plugins/security_solution/server/lib/detection_engine/rules/saved_object_mappings.ts

```diff
@@ -9,38 +9,84 @@ import { SavedObjectsType } from '../../../../../../../src/core/server';
 
 export const ruleStatusSavedObjectType = 'siem-detection-engine-rule-status';
 
-export const ruleStatusSavedObjectMappings: SavedObjectsType['mappings'] = {
+/**
+ * Records monitoring information about a rule, such as a when it last failed and why.
+ * This object is separate of the Rule SavedObject (why?)
+ * See `IRuleStatusSOAttributes` for types that related to this mapping
+ */
+const ruleStatusSavedObjectMappings: SavedObjectsType['mappings'] = {
   properties: {
+    /**
+     * A reference to a saved object
+     */
     alertId: {
       type: 'keyword',
     },
+    /**
+     * A `JobStatus`
```

Comment:
Not sure if there's a way to link this to the associated enum, but that would be nice.

Reply:
You can specify:
```suggestion
     * A string indicating the job's state
     * @type {JobStatus}
```

to at least hint at the enum, here. However, between the inline comments here and the explicit type above, these comments aren't visible to consumers of this object.